src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -1389,41 +1378,39 @@ protected String getSuccessfulDestination() {
 
     protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {}
 
-    // TODO boilerplate like this should not be necessary: JENKINS-22936
-    @Restricted(DoNotUse.class)
-    @RequirePOST
-    public void doDoRename(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isNameEditable() {
+        return true;
+    }
 
-        if (!hasPermission(CONFIGURE)) {
-            // rename is essentially delete followed by a create
-            checkPermission(CREATE);
-            checkPermission(DELETE);
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void checkRename(String newName) {
+        for (Job<?,?> job : getAllJobs()) {
+            if (job.isBuilding()) {
+                throw new Failure("Unable to rename a folder while a job inside it is building.");
+            }
         }
 
-        String newName = req.getParameter("newName");
-        Jenkins.checkGoodName(newName);
-
         String blocker = renameBlocker();
         if (blocker != null) {
-            rsp.sendRedirect("rename?newName=" + URLEncoder.encode(newName, "UTF-8") + "&blocker=" + URLEncoder.encode(blocker, "UTF-8"));
-            return;
+            throw new Failure(blocker);
         }
-
-        renameTo(newName);
-        rsp.sendRedirect2("../" + newName);
     }
 
     /**
      * Allows a subclass to block renames under dynamic conditions.
      * @return a message if rename should currently be prohibited, or null to allow
+     * @deprecated Override {@link #checkRename} instead.
```

Comment:
Make sure you are verifying compatibility with the `ComputedFolder` override: if the call to this from `checkRename` is commented out, some test ought to fail (if not here then in some downstream plugin like `branch-api` or `workflow-multibranch`).

Reply:
…and after verifying this, best to update `ComputedFolder` to avoid the deprecation.

Reply:
Ok, I will verify the compatibility and then update `ComputedFolder`.

Reply:
I did not see any downstream tests that appeared to test a rename, so I added a new test here to verify that the override worked before replacing `#renameBlocker` with `#checkRename`. See [ComputedFolderTest#renameWhileComputing](https://github.com/dwnusbaum/cloudbees-folder-plugin/blob/9d31f95c02a305787be69a26687d2a20254ea092/src/test/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolderTest.java#L486).